### ISSUE TITLE
Research (and possibly implement) other classifiers

### ISSUE THREAD

**Issue by bentsherman** [CUFCTL/face-recognition#23]
So the high-level problem that we are dealing with in this project is [statistical classification](https://en.wikipedia.org/wiki/Statistical_classification): given a training set of observations (images) with labels (identity of the person in the image), we are trying to determine the correct label of a new observation (test image); in other words, we are trying to "classify" the test image.

There are many different types of classifiers (see the Wikipedia article), but we currently use only one, [k-nearest neighbors](https://en.wikipedia.org/wiki/K-nearest_neighbors_algorithm). We use PCA, LDA, and ICA to (1) reduce the dimensionality and (2) extract important features from the images so that our classifier is faster and more accurate, but these three algorithms each use the same classifier. Some other classifiers that may be worth exploring:

- Support vector machines (SVMs)
- Bayes classifier
- Multinomial logistic regression (Softmax)
- Neural networks

This may be more of a long-term endeavor, depending on how many contributors we have. The priority is to test our current system against several datasets and tune the hyperparameters, but we could probably at least have a couple of people researching these techniques and thinking about (1) how useful they could be to us and (2) how we might implement them in our code.

Further reading: http://cs231n.github.io/classification/
Further reading: http://face-rec.org/algorithms/

**Comment by bentsherman** [comment]
Since all of these classifiers are implemented in scikit-learn, if we ever want to play around with them then we can do so in scikit-learn first.
